### DISCUSSION TITLE
I have an issue installing  smartcontractkit/chainlink-brownie-contracts

### DISCUSSION THREAD

**Discussion by tobezhanabi** [Cyfrin/foundry-full-course-cu#1192] [Q&A]
This is the error i got 
```
forge install smartcontractkit/chainlink-brownie-contracts@0.8.0 --no-commit
Installing chainlink-brownie-contracts in /home/tobez/foundry-23/fundMe-foundry23/lib/chainlink-brownie-contracts (url: Some("https://github.com/smartcontractkit/chainlink-brownie-contracts"), tag: Some("0.8.0"))
Cloning into '/home/tobez/foundry-23/fundMe-foundry23/lib/chainlink-brownie-contracts'...
remote: Enumerating objects: 32923, done.
remote: Counting objects: 100% (2524/2524), done.
remote: Compressing objects: 100% (1050/1050), done.
error: RPC failed; curl 56 GnuTLS recv error (-54): Error in the pull function.
fatal: the remote end hung up unexpectedly
fatal: early EOF
fatal: index-pack failed
fatal: clone of 'https://github.com/smartcontractkit/chainlink-brownie-contracts' into submodule path '/home/tobez/foundry-23/fundMe-foundry23/lib/chainlink-brownie-contracts' failed
Error: 
git submodule exited with code 128
```
I have tried
- Network reset
- git config --global http.postBuffer 524288000
- git gc --aggressive
-  Changed file and redo forge init
And finally updated my git version and fix some but now im stuck with this
```
error: RPC failed; curl 18 transfer closed with outstanding read data remaining
error: 1519 bytes of body are still expected
17:07:36.233535 pkt-line.c:85           packet:          git> 0002
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output
fatal: clone of 'https://github.com/smartcontractkit/chainlink-brownie-contracts' into submodule path '/home/tobez/foundry-23/fundMe-foundry23/lib/chainlink-brownie-contracts' failed
Error:
```
i even tried this-
set GIT_TRACE=1
set GIT_CURL_VERBOSE=1

how do i install the brownie please

**Comment by xyz899** [comment]
Hi,

Try these following things:
- Disable If there is a VPN when fetching data, or try to change location 
- Check if git and curl are up to date
- Verify that you have a stable connection 
- Restart your system, sometimes it can be due to the network cache

**Reply by tobezhanabi** [reply]
i have tried all, still nothing

**Reply by xyz899** [reply]
Check the answer [here](https://stackoverflow.com/questions/52529639/gnutls-recv-error-110-the-tls-connection-was-non-properly-terminated/52530617#52530617)](https://stackoverflow.com/questions/52529639/gnutls-recv-error-110-the-tls-connection-was-non-properly-terminated/52530617#52530617) : 

I know this has to do with git so find a way to remove git from your machine and reinstall the latest version there is using the executable (.exe) not a command line

If this does not work last resort is you increase http.postBuffer to a high value you have for example : `git config --global http.postBuffer 524288000 `

Check git docs for more info

**Reply by tobezhanabi** [reply]
I have tried all these. I honestly don't know why it keeps breaking but i will retry.
Thank you

**Reply by xyz899** [reply]
I apologize to ask but are you really sure you tried every possible scenarios I have gave you ? I am confident that there is a 95% chance your error is from here. Have you tried to forge init a new folder and install chainlink brownie contracts ?

**Reply by tobezhanabi** [reply]
I took your advice seriously and went out of my way to get my internet checked. Yes, it was the internet connection. An expert had to get involved. Thank u

**Reply by iBrainiac** [reply]
hello mine is related to this error code but when i try installing openzeppelin via forge. i Have a stable internet connection though

<img width="1089" alt="openzeppelin error" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/68105241/6b7cd964-5147-4b7c-8937-dad35cd589f3">
id like help thanks

**Reply by xyz899** [reply]
> hello mine is related to this error code but when i try installing openzeppelin via forge. i Have a stable internet connection though
> 
> 
> 
> <img width="1089" alt="openzeppelin error" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/68105241/6b7cd964-5147-4b7c-8937-dad35cd589f3">
> 
> id like help thanks

Have you configured your username and email address on Git Cli ? with git global. If you didn't that's probably why and if not check if your ssh keys are active and not revoked.

**Reply by c0deg33k** [reply]
https://github.com/Cyfrin/foundry-full-course-f23/discussions/1192?sort=new#discussioncomment-9179253

**Comment by c0deg33k** [comment]
I had the same issue. I tried some fixes recommended
i.e.
```
git submodule sync
```
```
git submodule update --init --recursive
```
Increased git buffer size
```
git config --global http.postBuffer 524288000
```
And lastly, tried to clone the repo inside the lib directory
```
https://github.com/smartcontractkit/chainlink-brownie-contracts.git
```
all to no avail. 

The culprit was my internet connection. My link despite being fast, was fluctuating and connection was dropping midway. I also noticed the connection speed would start out high then drop as it progressed.

Try changing your connection if you have a secondary link or speak to your ISP to get it resolved.

**Comment by FrankezeCode** [comment]
I had the same issue,  the problem is Internet 🛜 connection, just retry when the network is stable. That was how I resolved mine

**Reply by EngrPips** [reply]
Interesting

**Comment by 0xDelvine** [comment]
For the course, you just require the AggregatorV3Interface.sol in the smartcontractkit/chainlink-brownie-contracts so a simple fix would just be:
1. ``` cd /path/to/lib```
2. ```mkdir  AggregatorV3Interface && cd AggregatorV3Interface```
3. ```touch AggregatorV3Interface.sol```
4. Copy the `AggregatorV3Interface.sol`  code from [chainlink's smartcontractkit](https://github.com/smartcontractkit/chainlink-brownie-contracts/blob/main/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol) and paste it in the `AggregatorV3Interface.sol` text file you created inside the `AggregatorV3Interface` folder.

**Comment by EngrPips** [comment]
Interesting.
